crates/egui_plot/src/transform.rs

```diff
@@ -240,26 +252,51 @@ pub struct PlotTransform {
 }
 
 impl PlotTransform {
-    pub fn new(frame: Rect, mut bounds: PlotBounds, x_centered: bool, y_centered: bool) -> Self {
-        // Make sure they are not empty.
-        if !bounds.is_valid_x() {
-            bounds.set_x(&PlotBounds::new_symmetrical(1.0));
-        }
-        if !bounds.is_valid_y() {
-            bounds.set_y(&PlotBounds::new_symmetrical(1.0));
-        }
+    pub fn new(frame: Rect, bounds: PlotBounds, x_centered: bool, y_centered: bool) -> Self {
+        let mut new_bounds = bounds;
```

Comment:
alternative 🤷 

```suggestion
    pub fn new(frame: Rect, mut bounds: PlotBounds, x_centered: bool, y_centered: bool) -> Self {
```

Reply:
Nope! 😱  I need the original, unmodified `bounds` throughout, otherwise the modifications for the X axis interact badly with the modifications for the Y axis. Took me the longest time to figure that out.